### ISSUE TITLE
core.match support

### ISSUE THREAD

**Issue by tatut** [clj-kondo/clj-kondo#496]
In the example below clj-kondo warns "unresolved symbol a". 

```clojure 
(let [v [1 2 3]]
  (m/match v
           [a _ _] (println "first number is " a)
           :else (println "got something other than vector of three")))
```

**Comment by hjhamala** [comment]
This should also support core matches or-syntax which currently causes arity error for keyword -> 

(let [x 4 y 6 z 9]
  (match [x y z]
    [(:or 1 2 3) _ _] :a0
    [4 (:or 5 6 7) _] :a1
    :else nil))

**Comment by tatut** [comment]
My current plan is to just not lint the pattern part which is its own mini-language.
Only extract bindings from it so that they can be resolved in the consequence part.

**Comment by filipesilva** [comment]
Note: you can follow https://github.com/borkdude/clj-kondo/blob/master/doc/config.md#exclude-unresolved-symbols-from-being-reported to prevent those symbols from being reported for now.

**Comment by mikkosuonio** [comment]
Is there something I could do now to prevent an error from the or-keyword?

I get
```
match-test.cljs:6:14: error: keyword :or is called with 3 args but expects 1 or 2
match-test.cljs:6:26: error: unresolved symbol _
match-test.cljs:7:16: error: keyword :or is called with 3 args but expects 1 or 2
```
from
```
(ns match-test
  (:require [cljs.core.match] :as match))

(let [x 4 y 6 z 9]
     (match/match [x y z]
            [(:or 1 2 3) _ _] :a0
            [4 (:or 5 6 7) _] :a1
            :else nil))
```
using
```
{:linters {:unresolved-symbol   {:exclude [(clojure.core.match/match)
                                           (cljs.core.match/match)]}}}
```

**Comment by borkdude** [comment]
@mikkosuonio Yes, there is a typo in your require spec:

```
(ns match-test
  {:clj-kondo/config '{:linters {:unresolved-symbol {:exclude [(cljs.core.match/match)]}
                                 :invalid-arity {:skip-args [cljs.core.match/match]}}}}
  (:require [cljs.core.match :as match]))

(let [x 4 y 6 z 9]
  (match/match [x y z]
               [(:or 1 2 3) _ ] :a0
               [4 (:or 5 6 7) _] :a1
               :else nil))
```

**Comment by mikkosuonio** [comment]
Thanks for pointing that out!

Config for invalid-arity with skip-args seems to work fine!

**Comment by borkdude** [comment]
Work in progress in the `core-match` branch:

<img width="341" alt="Screenshot 2021-01-26 at 23 30 16" src="https://user-images.githubusercontent.com/284934/105914475-b6688f80-602e-11eb-9ef5-7b86ccde987c.png">

**Comment by borkdude** [comment]
Merged part 1 to master now. Awaiting your feedback.

You can grab a binary from CircleCI / appveyor or use the JVM version.

**Comment by borkdude** [comment]
And now the final part, part 2 is merged too!
